node/node.go

```diff
@@ -1394,10 +1367,57 @@ func (app *App) Start(ctx context.Context) error {
 			app.host.ID().String(), app.Config.Genesis.GenesisID().ShortString())
 	}
 
-	if err := app.startServices(ctx, appErr); err != nil {
+	if err := app.startServices(ctx); err != nil {
 		return err
 	}
 
+	// need post verifying service to start first
+	if preserve != nil {
+		for i, poetProof := range preserve.Proofs {
+			encoded, err := codec.Encode(poetProof)
+			if err != nil {
+				app.log.With().Error("failed to encode poet proof after checkpoint",
+					log.Stringer("atx id", preserve.Deps[i].ID()),
+					log.Object("poet proof", poetProof),
+					log.Err(err),
+				)
+			} else {
+				if err := app.poetDb.ValidateAndStoreMsg(ctx, p2p.NoPeer, encoded); err != nil {
+					app.log.With().Error("failed to preserve poet proof after checkpoint",
+						log.Stringer("atx id", preserve.Deps[i].ID()),
+						log.String("poet proof ref", preserve.Deps[i].GetPoetProofRef().ShortString()),
+						log.Err(err),
+					)
+				} else {
+					app.log.With().Info("preserved poet proof after checkpoint",
+						log.Stringer("atx id", preserve.Deps[i].ID()),
+						log.String("poet proof ref", preserve.Deps[i].GetPoetProofRef().ShortString()),
+					)
+				}
+			}
+		}
```

Comment:
Same as the other, `continue` helps to prevent deep nesting.

```suggestion
		for i, poetProof := range preserve.Proofs {
			encoded, err := codec.Encode(poetProof)
			if err != nil {
				app.log.With().Error("failed to encode poet proof after checkpoint",
					log.Stringer("atx id", preserve.Deps[i].ID()),
					log.Object("poet proof", poetProof),
					log.Err(err),
				)
				continue
			}
			if err := app.poetDb.ValidateAndStoreMsg(ctx, p2p.NoPeer, encoded); err != nil {
				app.log.With().Error("failed to preserve poet proof after checkpoint",
					log.Stringer("atx id", preserve.Deps[i].ID()),
					log.String("poet proof ref", preserve.Deps[i].GetPoetProofRef().ShortString()),
					log.Err(err),
				)
				continue
			}
			app.log.With().Info("preserved poet proof after checkpoint",
				log.Stringer("atx id", preserve.Deps[i].ID()),
				log.String("poet proof ref", preserve.Deps[i].GetPoetProofRef().ShortString()),
			)
		}
```

Reply:
moved to a separate function and used continue.